sdk/storage/Azure.Storage.Blobs/src/BlobUriBuilder.cs

```diff
@@ -215,7 +216,7 @@ public BlobUriBuilder(Uri uri)
                 else
                 {
                     BlobContainerName = path.Substring(startIndex, containerEndIndex - startIndex); // The container name is the part between the slashes
-                    BlobName = path.Substring(containerEndIndex + 1);   // The blob name is after the container slash
+                    BlobName = path.Substring(containerEndIndex + 1).UnescapePath();   // The blob name is after the container slash
```

Comment:
Will this function be called in other API?
If so, please cover other API's test with blob name contains special chars to make sure no regression.

Reply:
This is covered by previously written tests - https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Blobs/tests/ContainerClientTests.cs#L2256